framework/PostProcessors/ETimporter.py

```diff
@@ -0,0 +1,262 @@
+
+"""
+
+from __future__ import division, print_function , unicode_literals, absolute_import
+import warnings
+warnings.simplefilter('default', DeprecationWarning)
+
+#External Modules---------------------------------------------------------------
+import numpy as np
+import xml.etree.ElementTree as ET
+#External Modules End-----------------------------------------------------------
+
+#Internal Modules---------------------------------------------------------------
+from .PostProcessor import PostProcessor
+from utils import InputData
+import Files
+import Runners
+#Internal Modules End-----------------------------------------------------------
+
+
+class ETimporter(PostProcessor):
+  """
+    This is the base class for postprocessors
+  """
+  def __init__(self, messageHandler):
+    """
+      Constructor
+      @ In, messageHandler, MessageHandler, message handler object
+      @ Out, None
+    """
+    PostProcessor.__init__(self, messageHandler)
+    self.printTag = 'POSTPROCESSOR ET IMPORTER'
+    self.ETformat = None
+    self.allowedFormats = ['OpenPSA']
+
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = super(ETimporter, cls).getInputSpecification()
+
+    return inputSpecification
+
+
+  def initialize(self, runInfo, inputs, initDict) :
+    """
+      Method to initialize the pp.
+      @ In, runInfo, dict, dictionary of run info (e.g. working dir, etc)
+      @ In, inputs, list, list of inputs
+      @ In, initDict, dict, dictionary with initialization options
+      @ Out, None
+    """
+    PostProcessor.initialize(self, runInfo, inputs, initDict)
+    self.inputs = inputs
+    self._workingDir = runInfo['WorkingDir']
+
+  def _localReadMoreXML(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.Element, Xml element node
+      @ Out, None
+    """
+    for child in xmlNode:
+      if child.tag == 'fileFormat':
+        if child.text not in self.allowedFormats:
+          self.raiseAnError(IOError, 'ETimporterPostProcessor Post-Processor ' + self.name + ', format ' + child.text + ' : is not supported')
+        else:
+          self.ETformat = child.text
+      else:
+        self.raiseAnError(IOError, 'ETimporterPostProcessor Post-Processor ' + self.name + ', node ' + child.tag + ' : is not recognized')
+
+  def run(self, input):
+    """
+      This method executes the postprocessor action.
+      @ In,  input, object, object containing the data to process. (inputToInternal output)
+      @ Out, None
+    """
+    if self.ETformat == 'OpenPSA':
```

Comment:
this if statement is probably redundant with the current implementation since the OpenPSA is the only one we support (and you check for it before).
If you want to keep it, I would use a lambda function with a dictionary mapping